modules.cmake

```diff
@@ -57,11 +124,12 @@ function(add_module_library name)
   add_library(${name})
   set_target_properties(${name} PROPERTIES LINKER_LANGUAGE CXX)
 
-  if (NOT ${${AML_IF}})
-    # Create a non-modular library.
+  # Add fallback sources to the target in case modules are not supported or fallback was explicitly selected
+  modules_supported(modules_available)
+  if (NOT ${modules_available} OR (DEFINED AML_IF AND NOT AML_IF))
```

Comment:
`AML_IF` should be checked first. This is used in {fmt} to avoid requiring both `FMT_MODULE` and setting the standard.

Reply:
fixed